vortex-expr/src/field_paths.rs

```diff
@@ -0,0 +1,64 @@
+use vortex_dtype::field_paths::FieldPath;
+
+use crate::expressions::{Predicate, Value};
+use crate::operators::Operator;
+
+pub trait FieldPathOperations {
```

Comment:
Unclear what you gain from this being a trait. It probably should all be methods on FieldPath struct